independent-projects/arc/runtime/src/main/java/io/quarkus/arc/ArcCDIProvider.java

```diff
@@ -37,6 +37,9 @@ public ArcCDIProvider() {
 
     @Override
     public CDI<Object> getCDI() {
+        if (Arc.container() == null) {
+            throw new IllegalStateException();
```

Comment:
Any chance you could add a real message?

This exception is caught at some point?

Reply:
shouldn't we pass some error message, just in case someone reads the log.

Reply:
Sure we could. I did not add anything because the contract of `CDI.current()` and `CDIProvider.getCDI()` is very clear.